### PR TITLE
Add basic button functionalities

### PR DESCRIPTION
**Pull request by kienvo**
What I did:

- Provided a way to set handler for each button one-press/long-press
- Debouncing with RC filter and Schmitt trigger

### CODE REVIEW

File: src/button.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#include "button.h"
+
+#define DEBOUNCE_HIGH_THRES (200) // 0-255
+#define DEBOUNCE_LOW_THRES  (55)  // 0-255
+#define LONGPRESS_THRES     (25)  // Hz
+#define BUTTON_SCAN_FREQ    (50)  // Hz
+
+#define IS_INRANGE(v, start, stop) ((v) > (start) && (v) < (stop))
```
Comment (fcartegnie) [inline src/button.c]: useless macro or needs to be changed (>= start) to apply to previous branch where it is used.

File: src/main.c [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+					(v) = (min)
 
-uint16_t fb[2][FB_WIDTH];
+enum MODES {
+	NORMAL = 0,
+	DOWNLOAD,
+	POWER_OFF,
+	MODES_COUNT,
+};
+#define BRIGHT_LEVELS   (4)
```
Comment (fcartegnie) [inline src/main.c]: BRIGHTNESS

File: src/main.c [hunk 3]
```diff
@@ -55,16 +93,31 @@ int main()
...
 
-	while (1) {
-	}
+	btn_init();
+	btn_onOnePress(KEY1, change_mode);
+	btn_onOnePress(KEY2, change_fb);
+	btn_onLongPress(KEY1, change_brightness);
+
+    while (1) {
+		int i = 0;
```
Comment (fcartegnie) [inline src/main.c:108]: unsigned

Comment (kienvo) [reply src/main.c:108]: you will need to hold the button for 13.6102208 years to get this int overflowed =]

**Pull request merged**